### ISSUE TITLE
Show the direct download link for the DietPi Bullseye image

### ISSUE THREAD

**Issue by char8x** [MichaIng/DietPi-Website#318]
As not all software in dietpi catalogue is compatible with Debian Bookworm yet, providing a direct download link is still necessary and I can submit a PR.

### Original
<img width="548" alt="图片" src="https://github.com/MichaIng/DietPi-Website/assets/6831013/f8fc528d-98c0-4a2f-aae7-a9dd30a77515">

### Modified
<img width="563" alt="图片" src="https://github.com/MichaIng/DietPi-Website/assets/6831013/1d257fa9-21ce-45dd-a7a1-9394398cc0f7">

**Comment by MichaIng** [comment]
We intentionally do not want to make it too easy do download Bulleye images, without at least reading through the blog post and understanding when/why it might be suitable, any why in general not.

Instead of just downloading Bullseye whenever facing any issue with Bookworm which might or might not be trivial to fix, it IMO makes much more sense to first check back with us whether/how to fix it for Bookworm, and in case communicate with the developers/maintainers of the incompatible software, to make it compatible with the soon 1 year released Debian version. Spreading Bullseye images too long creates an additional maintenance burden for us, and I actually wanted to remove and stop providing these images soon.

So I am clearly against it, but open for other arguments/opinions.

**Comment by char8x** [comment]
Certainly, from a maintainer's perspective, it is ideal to discover and resolve software compatibility issues as early as possible.

I'm not sure about the user base of DietPi, but it's possible that some users may prefer to report issues and wait for solutions, while others may simply need a Debian operating system that works immediately. For the latter group, providing a "just works" image like the Bullseye Image remains necessary.

If there's a consideration to remove the Bullseye image from the website:
- Would it be possible to wait until Bullseye enters the Long-Term Support (LTS) maintenance phase (scheduled for 2024-08-15) before removal?
- Could there be a supplement in the documentation explaining how users can build their own Bullseye image, such as cloning the DietPi repository and running their own GitHub Actions?
- Alternatively, it might be suggested that users utilize Bullseye images built by Armbian or the official Debian Bullseye images.

**Comment by Joulinar** [comment]
You simply can use our DietPi install script that will work on Bullseye still. Even if we remove the images from our download page.

**Comment by char8x** [comment]
@Joulinar Are you referring to the portion mentioned in [Script Execution](https://dietpi.com/docs/hardware/#script-execution), specifically the part `bash -c "$(curl -sSfL 'https://raw.githubusercontent.com/MichaIng/DietPi/master/.build/images/dietpi-installer')"`

**Comment by Joulinar** [comment]
Yes it's described on our online docs

**Comment by MichaIng** [comment]
You can also use our build script, but it is not so well documented: https://github.com/MichaIng/DietPi/blob/master/.build/images/dietpi-build
Or even the workflow on GitHub from your fork: https://github.com/MichaIng/DietPi/actions/workflows/dietpi-build.yml
You just need to adjust it to either upload elsewhere or store images as assets on GitHub directly, in case you are familiar with GitHub Actions.

However, official EOL (LTS) from Debian end is probably a good target to stop providing Bullseye images. The installer and build system can support it for another year or so. In general at least for container images, it needs to stay until we fully drop Bullseye support anyway, since we use these containers to compile software for Bullseye and test software installations.

>  others may simply need a Debian operating system that works immediately. For the latter group, providing a "just works" image like the Bullseye Image remains necessary.

Yes and no. Of course I understand that some want the things they know/are used to, to "just work". One of the main goals of DietPi is to provide software installation options which just work. But making the choice too easy for everyone to pick a dead end distribution version and probably get further used to software which is not maintained anymore, hence a potential security risk to use, making the step to switch to some modern well maintained alternative software harder later on, can be also in contradiction to the goal, after all. So yeah, naturally it is a balance act to find the right time for this. There are some particular software options I am also awaiting Bookworm/libssl3/PHP8 support for. These are also a good argument to keep the Bullseye images.

**Comment by char8x** [comment]
I took a brief look at the documentation, and for building a Dietpi image on my own, there are two scenarios:

- If I have an idle, non-daily use Debian system or a Debian-based virtual machine, I can use this `https://raw.githubusercontent.com/MichaIng/DietPi/master/.build/images/dietpi-installer` script.

- Alternatively, I can fork the Dietpi repository, make some adjustments to the `https://github.com/MichaIng/DietPi/actions/workflows/dietpi-build.yml` file, and then build it using Github Actions.

Regarding whether to display the download link, I think we can maintain the current situation. Feel free to close this issue.
